static/src/javascripts/lib/proximity-loader.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import bonzo from 'bonzo';
+import debounce from 'lodash/functions/debounce';
+import fastdom from 'lib/fastdom-promise';
+import mediator from 'lib/mediator';
+
+let items = [];
+const scroll = { top: 0, bottom: 0 };
+
+const doProximityLoading = (): void => {
+    scroll.top = window.pageYOffset;
+    scroll.bottom = scroll.top + bonzo.viewport().height;
+    items = items.filter(item => {
+        if (item.conditionFn()) {
+            item.loadFn();
+            return false;
+        }
+
+        return true;
+    });
+
+    if (items.length === 0) {
+        mediator.off('window:throttledScroll', doProximityLoading);
+    }
+};
+
+const addItem = (conditionFn: () => boolean, loadFn: () => void): void => {
+    // calls `loadFn` when `conditionFn` is true
+    const item = { conditionFn, loadFn };
+
+    items.push(item);
+
+    if (items.length === 1) {
+        mediator.on('window:throttledScroll', doProximityLoading);
+    }
+
+    const doProximityLoadingDebounced = debounce(doProximityLoading, 2000);
+    doProximityLoadingDebounced();
```

Comment:
I'm not sure how important, but this is a a change in behaviour. `debounce` forces the resulting function to wait at least 2000 milliseconds since the previous time it was invoked, before executing the passed function. Here a new debounced function is being created every time `addItem` is executed, so the `debounce` cooldown will be discarded after `addItem` has finished executing. 

Maybe it's not important in this instance. What do you think?

Reply:
After reading the lodash docs around `debounce` yesterday I agree with Simon's comment above :point_up:

Reply:
I agree with you Will change!